src/array.cr

```diff
@@ -1545,11 +1544,39 @@ class Array(T)
     self
   end
 
+  # Returns `self` with all the elements shifted `n` times.
+  #
+  # ```
+  # a = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  # a.rotate!    # => [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
+  # a.rotate!(1) # => [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
```

Comment:
Since `#rotate!` modifies `a`, this is actually the result of a 0 rotation.
```suggestion
  # a.rotate!(0) # => [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
```
And the next one rotates by 2.

You might want to choose different results, and maybe also showcase a negative rotation.

Reply:
This is 0 indexed so I think the return on `rotate(1)` would start with `1`

Reply:
Check the result to verify: https://carc.in/#/r/82k0

The first call to `#rotate!` modifies `a`, so the second one rotates the result from the first one.

Reply:
I see your point.  I have updated the docs.